tests/test_yaml_run_args/test_yaml_run_args.py

```diff
@@ -0,0 +1,227 @@
+                "searcher": "bayesian_optimization",
+                "searcher_kwargs": {"initial_design_size": 5, "surrogate_model": "gp"},
+            },
+        ),
+        (
+            "run_args_partial_same_level.yaml",
+            {
+                "root_directory": "test_yaml",
+                "max_evaluations_total": 20,
+                "overwrite_working_directory": True,
+                "post_run_summary": False,
+                "task_id": 4,
+                "continue_until_max_evaluation_completed": True,
+                "ignore_errors": True,
+            },
+        ),
+        ("run_args_empty.yaml", {}),
+        ("run_args_optional_loading_format.yaml", {
+            "run_pipeline": run_pipeline,
+            "pipeline_space": pipeline_space,
+            "root_directory": "test_yaml",
+            "max_evaluations_total": 20,
+            "max_cost_total": 4.2,
+            "overwrite_working_directory": True,
+            "post_run_summary": False,
+            "development_stage_id": 9,
+            "max_evaluations_per_run": 5,
+            "continue_until_max_evaluation_completed": True,
+            "loss_value_on_error": 2.4,
+            "cost_value_on_error": 2.1,
+            "ignore_errors": False,
+            "searcher": BayesianOptimization,
+            "searcher_path": "/path/to/searcher",
+            "searcher_kwargs": {
+                "initial_design_size": 5,
+                "surrogate_model": "gp"
+            },
+            "pre_load_hooks": [hook1]
+
+        })
+    ],
+)
+def test_yaml_config(yaml_path: str, expected_output: Dict) -> None:
+    """
+    Tests NePS configuration loading from run_args=YAML, comparing expected settings
+    against loaded ones. Covers hierarchical levels and partial/full of yaml
+    dict definitions.
+
+    Args:
+        yaml_path (str): Path to the YAML file.
+        expected_output (dict): Expected configuration settings.
+    """
+    check_run_args(yaml_path, expected_output)
+
+
+@pytest.mark.neps_api
+@pytest.mark.parametrize(
+    "yaml_path, expected_exception",
+    [
+        ("run_args_invalid_type.yaml", TypeError),
+        ("run_args_wrong_path.yaml", ImportError),
+        ("run_args_invalid_key.yaml", KeyError),
+        ("run_args_wrong_name.yaml", ImportError),
+        ("run_args_key_missing.yaml", KeyError),
+    ],
+)
+def test_yaml_failure_cases(yaml_path: str, expected_exception: Type[Exception]) -> None:
+    """
+    Tests for expected exceptions when loading erroneous NePS configurations from YAML.
+
+    Each case checks if `get_run_args_from_yaml` raises the correct exception for errors
+    like invalid types, missing keys, and incorrect paths in YAML configurations.
+
+    Args:
+        yaml_path (str): Path to the error-containing YAML file.
+        expected_exception (Exception): Expected exception type.
+    """
+    with pytest.raises(expected_exception):
+        get_run_args_from_yaml(BASE_PATH + yaml_path)
```

Comment:
Save yourself some effort as I believe the code itself is pretty explanatory and the docstring doesn't add any new information than isn't seen directly from reading the code.